### DISCUSSION TITLE
Which app am I in?

### DISCUSSION THREAD

**Discussion by aral** [elementary/gala#982] [General]
## Problem

![Screenshot from 2020-12-06 14-18-01@2x](https://user-images.githubusercontent.com/41798/101282641-2fd1c780-37ce-11eb-9c82-80fc15b8439b.png)

From the above screenshot, can someone please tell me which app I’m currently running and let me know how you managed to figure it out.

## Proposal

Display the name of the currently-running app.

One place where it would make sense is the status display at the middle of the of the top bar. It shows the current date and time, it would make sense to show the current app there too. e.g., 

```
🔍️ Applications                      Epiphany    Sun, Dec 6    14:24                     [icons]
```

## Prior Art

Knowing which app you’re currently in is a commonly implemented example of a basic landmark.

(Apologies for the huge screenshot, I’m posting this from elementary OS and I couldn’t find a crop tool in the Photos app.)

**Comment by cassidyjames** [comment]
It looks like Epiphany is focused here because it has the active window styling, which is higher contrast. The elementary stylesheet has a clear distinction between focused and unfocused windows—even moreso in the redesign coming for elementary OS 6; I think part of the problem here is that Firefox is not a native app and thus does not fully use the native stylesheet, and thus misses out on the nuances of that affordance. If Firefox used the native toolkit, it would be much more clear. This is a huge reason we are in favor of native apps that care about the platform on which they run, because we can insure these sort of affordances at the platform level.

Regarding your suggestion of showing the app name in the panel, I'm unsure that's a real solution for most situations. Multi-window apps are common, and showing the app name does not specify which window is focused. Trying to show more context (like a focused web page’s title or a focused document name) could work, but then that string quickly gets very long and dominates the entire panel, or must be truncated and lose context anyway.

Our chosen solution to this issue is to lean on the stylesheet. If it can be made more clear in the stylesheet, that's a stylesheet issue. If an app chooses not to use the platform stylesheet, there is only so much we can do while still optimizing for apps that _do_ care about the platform.

**Comment by aral** [comment]
But how do you know it’s Epiphany? And not Firefox?

**Comment by aral** [comment]
(My point isn’t which app is focused. I’ve never seen either app before. How do I know which is Epiphany and which is Firefox?)

**Comment by danirabbit** [comment]
Okay, counterpoint: which app is this? :p

![image](https://user-images.githubusercontent.com/7277719/101296062-3c5a1e00-37d6-11eb-8d75-2f950b123756.png)

Presumably, you installed and opened these apps so I would assume you have knowledge of what you have done. But either way, does it really matter if you can see the content and interact with it and do the thing you set out to do?

**Comment by aral** [comment]
This is the equivalent screenshot from iOS:

![5F906CFF-8A27-41D6-92B9-52016605934F](https://user-images.githubusercontent.com/41798/101296468-be206a80-381b-11eb-868b-e427a0542e9e.png)

In a multi-window environment, not knowing which app is currently active is a basic usability flaw. You’re missing the most important landmark: where am I now?

(A more equivalent example would be the app name as displayed on macOS but we shouldn’t even need that as a reference as it can be derived from first principles.)

**Reply by martinstork** [reply]
Would a visual indicator in the multitasking view be a fitting solution? _A tinted border around the active window for example._
![BA53882C-67CA-4D9C-9D4B-ACD47A01CB37](https://user-images.githubusercontent.com/15929191/103405974-4cf18000-4b59-11eb-925a-00d9af5646d5.png)

**Comment by cassidyjames** [comment]
Just following up as this has seemed to be a multi-faceted issue; and to clarify, closing it doesn't mean we're not addressing those concerns. 

I've proposed https://github.com/elementary/stylesheet/pull/899 to further improve "backdrop" (unfocused window) distinction. We do show app icons in the multitasking view, and have plans to further improve the Alt+Tab window switching view to show window titles (https://github.com/elementary/gala/discussions/72).

It might make sense to convert this into a discussion since it's not a specific issue, but touches on several things.

**Comment by jlnr** [comment]
I'm not on the UX team, but it annoys me when I cannot tell one app from another without thinking. It still happens sometimes when I have several different JetBrains IDEs open, and I also had it happen on iOS back when iMessage had no avatars and looked the same as Mail. So I agree with the stated problem.

However, I don't agree with the proposed solution because the label would always be slightly off to the left, even when the right app is focused. In my experience, I find it much easier to orient myself when apps go all-in on their color scheme, like the Ephemeral browser on elementary, or Microsoft's color-coded office apps on macOS. That doesn't help new users, but if they really get lost they can click on a dock icon again or use one of the overviews to find out which window belongs to which app.

**Comment by pushqrdx** [comment]
why not just add the app name in place of the kinda useless Applications label in the wingpanel :D

**Comment by hanaral** [comment]
Here are a couple of ideas;

How about just having Gala add a 'Show X in dock' entry with a colour icon to the top of titlebar context menus? Clicking it could make the app bounce a couple of times.

Another idea is to get rid of this concept that the dock is an application launcher rather than a part of the multi-tasking experience by making focused apps rise slightly? I don't quite know why the focused app highlight went away when it's obviously not possible for all apps to distinguish themselves purely with colour. I can think of 2 appcenter apps and another elementary app that uses the Grape colour for their icon and app primary colour; are you just going to tell other developers to go and find another colour to use?

Another option is of course to use regardless of these choices is to add an accel label when hovering on a titlebar, but this assumes that you are using cursor-based navigation and not switch-based.
